.travis.yml

```diff
@@ -73,6 +73,7 @@ jobs:
     - git fetch --unshallow origin master:master
     script:
     - tools/ppa-upload.sh
+    - lftp -e "help;quit" ppa.launchpad.net
```

Comment:
This doesn't actually seem to reach out to `ppa.launchpad.net`, just return the help?

Reply:
@Saviq my understanding is that it will open an FTP connection with ppa.launchpad.net and then I just want to get some interactions in the FTP session and return. I'm open to suggestions on how to make a more thorough check.

Reply:
It doesn't even make the connection, though:

```
$ lftp -e "help;quit" google.com
    !<shell-command>                     (commands)       
    alias [<name> [<value>]]             attach [PID]             
[...]
# command exits with 0
```
vs.
```
$ lftp -e "set net:timeout 5;set net:max-retries 2;ls;quit" google.com
ls: Fatal error: max-retries exceeded
# exits with error
```

Reply:
@Saviq Just curious, how can you tell that it's not making a connection? It's calling `help` *in the FTP session* and then exiting the session with `quit`.

You can try it manually:

```
$ lftp ppa.launchpad.net
lftp ppa.launchpad.net:~> help
    !<shell-command>                     (commands)                           alias [<name> [<value>]]             attach [PID]
[...]
lftp ppa.launchpad.net:~> quit
```

Reply:
I've updated the FTP commands with what you have suggested.

Reply:
@cotsog by the fact `google.com` is not an ftp server, and I get the help output and no complaint. Basically what I've shown above.

`help` is not an FTP protocol command, it's a `lftp` command that it handles internally, and it doesn't even reach out to the server until it gets a command that actually requires a connection.

Just replace `ppa.launchpad.net` with google in your example above and you'll see it's happy, even though `google.com` drops packets on port 21.